common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -5684,6 +5684,11 @@ public static enum ConfVars {
             + "  recompile_without_cbo: recompiles query after a CBO failure\n"
             + "  reexecute_lost_am: reexecutes query if it failed due to tez am node gets decommissioned\n "
             + "  write_conflict: retries the query once if the query failed due to write_conflict"),
+    HIVE_QUERY_CUSTOM_REEXECUTION_STRATEGIES("hive.query.custom.reexecution.strategies",
```

Comment:
not sure if we need both configs. I think we can create a generic config `hive.query.reexecution.strategies` and list existing strategies there similar to `metastore.task.threads.remote`
````
org.apache.hadoop.hive.ql.reexec.ReExecutionOverlayPlugin,
org.apache.hadoop.hive.ql.reexec.ReExecuteLostAMQueryPlugin,
org.apache.hadoop.hive.ql.reexec.ReCompileWithoutCBOPlugin,
org.apache.hadoop.hive.ql.reexec.ReExecutionDagSubmitPlugin,
org.apache.hadoop.hive.ql.reexec.ReExecuteOnWriteConflictPlugin;
````

Reply:
or maybe have mix of enum defined strategies and custom classes
````
overlay,reoptimize,reexecute_lost_am,dagsubmit,recompile_without_cbo,write_conflict,org.apache.hadoop.hive.ql.reexec.custom.CustomPlugin1
````

Reply:
I see. I will do that.

Reply:
@deniskuzZ  I fixed that.